test/unit/puppetlabs/services/ca/certificate_authority_core_test.clj

```diff
@@ -119,6 +119,44 @@
           response (ring-app request)]
       (is (= version-number (get-in response [:headers "X-Puppet-Version"]))))))
 
+(deftest handle-delete-certificate-request!-test
+  (let [settings      (assoc (testutils/ca-sandbox! cadir)
+                             :allow-duplicate-certs true
+                             :autosign false)
+        subject       "foo-agent"
+        csr-stream    (io/input-stream (test-pem-file "foo-agent-csr.pem"))
+        expected-path (ca/path-to-cert-request (:csrdir settings) subject)
+        assert-proper
+          (fn
```

Comment:
We have a strong tendency to just duplicate the assertions in the test instead of putting them in a helper function like this.  It tends to make the tests longer and more repetitive but much easier to debug when something breaks.